src/visualizers-workbench/P5VisualizerTemplate.ts

```diff
@@ -0,0 +1,174 @@
+import {P5Visualizer} from '../visualizers/P5Visualizer'
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+
+/** md
+# p5 Visualizer Template
+
+[image should go here]
+
+This template can be used as a starting point for designing a p5 visualizer. You
+can use it to step back and forth through a sequence, showing each element as a
+written number.
+
+## Parameters
+
+- Step size
+
+## Interactions
+
+- Use the left and right arrow keys to step
+**/
+
+class P5VisualizerTemplate extends P5Visualizer {
+    // the name that appears in bundle card titles
+    static visualizationName = 'p5 Visualizer Template'
+
+    // external parameters, which the user can choose while creating the
+    // visualizer bundle
+    stepSize = 1 // the default value of this parameter is zero
+    params = {
+        /** md
+- stepSize: How far to step when the user presses a left or right arrow key.
+         **/
+        stepSize: {
+            value: this.stepSize,
+            forceType: 'integer',
+            displayName: 'Step size',
+            required: true,
+        },
+    }
+
+    // internal parameters. the value of `index` will always be overwritten in
+    // `inhabit()`, but TypeScript can't infer that, so we have to give `index`
+    // an initial value
+    index = 0
+    justStepped = false
+    flash = 0
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        // make sure the step size is positive
+        if (this.params.stepSize.value <= 0) {
+            status.isValid = false
+            status.errors.push('Step size must be positive')
+        }
+
+        return status
+    }
+
+    inhabit(element: HTMLElement): void {
+        // do the behind-the-scenes work of managing p5 canvases. this work is
+        // finicky, and your visualizer won't work unless it's done properly, so
+        // you should leave it to the superclass unless you have a very
+        // compelling reason to do it yourself
+        super.inhabit(element)
+    }
```

Comment:
It is not good practice to define a method such that all it does is call the super.method() so we shouldn't model that in the template. Personally I think that since needing to write an inhabit() seems like it is going to be quite rare, we could leave it out of the example altogether. It's not as though we put in a depart() method, even though a Visualizer writer could extend that to display a popup saying "goodbye I am going to miss you" if they really wanted to.

Reply:
Whoops! Removed.